File: PDP/Odev2/KISIM1/Odev2Kutuphane/src/RandomPacket/Kisi.java
/**
*
* @author Melih Ensar BARIŞIK ensar.barisik@ogr.sakarya.edu.tr / Ömer Can ÇALIŞIR omer.calisir@ogr.sakarya.edu.tr
* @since  25.04.2020
* <p>
*   Kişi oluşturulan kısım 
* </p>
*/
package RandomPacket;


/**
 *
 * @author bar_i
 */
public class Kisi {
    
  
    private KimlikNo kimlikNo;
    private String isim_soyisim;
    private int yas;
    private Telefon telefonNo;

    public KimlikNo getKimlikNo() {
        return kimlikNo;
    }

    public void setKimlikNo(KimlikNo kimlikNo) {
        this.kimlikNo = kimlikNo;
    }

    public String getIsim_soyisim() {
        return isim_soyisim;
    }

    public void setIsim_soyisim(String isim_soyisim) {
        this.isim_soyisim = isim_soyisim;
    }

    public int getYas() {
        return yas;
    }

    public void setYas(int yas) {
        this.yas = yas;
    }

    public Telefon getTelefonNo() {
        return telefonNo;
    }

    public void setTelefonNo(Telefon telefonNo) {
        this.telefonNo = telefonNo;
    }

    //Dışarıdan alınan nesneleri ve değişkenleri nesne referanslarına ve diğer değişkenlere atama yapan kurucu fonksiyon
    public Kisi(KimlikNo kimlikNo, String isim, int yas, Telefon telefonNo) {
        this.kimlikNo = kimlikNo;
        this.isim_soyisim = isim;
        this.yas = yas;
        this.telefonNo = telefonNo;
    }


    
    
}

File: PDP/Odev2/KISIM2/src/Kisi.c
/**
*
* @author Melih Ensar BARIŞIK ensar.barisik@ogr.sakarya.edu.tr / Ömer Can ÇALIŞIR omer.calisir@ogr.sakarya.edu.tr
* @since  25.04.2020
* <p>
*   Kişi oluşturulan kısım 
* </p>
*/
#include "Kisi.h"
#include "stdio.h"
#include "stdlib.h"

// nesne yönelimli benzetme için sözde kurucu fonksiyonu
Kisi KisiOlustur(KimlikNo kimlikNo,char* isim_soyisim,int yas,Telefon telefonNo)
{
	Kisi this;
	// struct için bellekte yer ayırma işlemi
	this=(Kisi)malloc(sizeof(struct KISI));
	
	// dışarıdan alınan, kişiyi oluşturan parametrelerin
	// atanması işlemi
	this->kimlikNo=kimlikNo;
	this->isim_soyisim=isim_soyisim;
	this->yas=yas;
	this->telefonNo=telefonNo;



	// fonksiyon göstericisine oluşturulan
    // fonksiyonun adresinin verilmesi işlemi
	this->yokediciKisi=&_yokediciKisi;


	return this;
	
}

// bellekten alınan yerin geri teslimi için yıkıcı fonksiyon
void _yokediciKisi(Kisi this)
{
	if(this==NULL) return;
	free(this);
}
